src/daemon/bitcoind/interface.rs

```diff
@@ -230,6 +230,16 @@ impl BitcoinD {
         )))
     }
 
+    pub fn unloadwallet(&self, wallet_path: String) -> Result<(), BitcoindError> {
+        let res = self.make_node_request("unloadwallet", &params!(Json::String(wallet_path),))?;
+
+        if let Some(warning) = res.get("warning").map(|s| s.to_string()) {
```

Comment:
You have to check if warning is different from `""`. From the docs:
```
bitcoin-cli help unloadwallet 
unloadwallet ( "wallet_name" load_on_startup )
Unloads the wallet referenced by the request endpoint otherwise unloads the wallet specified in the argument.
Specifying the wallet name on a wallet endpoint is invalid.
Arguments:
1. wallet_name        (string, optional, default=the wallet name from the RPC endpoint) The name of the wallet to unload. Must be provided in the RPC endpoint or this parameter (but not both).
2. load_on_startup    (boolean, optional, default=null) Save wallet name to persistent settings and load on startup. True to add wallet to startup list, false to remove, null to leave unchanged.

Result:
{                       (json object)
  "warning" : "str"     (string) Warning message if wallet was not unloaded cleanly.
}
```
So `warning` is always there, and if everything is fine it's equal to `""`

Reply:
Oh, right, thanks

Reply:
Fixed